Core/Core/Planner/CreateToDo/ViewModel/CreateToDoViewModel.swift

```diff
@@ -0,0 +1,136 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+import SwiftUI
+
+final class CreateToDoViewModel: ObservableObject {
+
+    // MARK: - Output
+
+    let pageTitle = String(localized: "New To Do", bundle: .core)
+    let pageViewEvent = ScreenViewTrackingParameters(eventName: "/calendar/new")
+    let screenConfig = InstUI.BaseScreenConfig(refreshable: false)
+
+    @Published private(set) var state: InstUI.ScreenState = .data
+    @Published var title: String = ""
+    @Published var date: Date?
+    @Published private var calendar: CDCalendarFilterEntry?
+    @Published var details: String = ""
+    @Published var shouldShowAlert: Bool = false
+
+    var isAddButtonEnabled: Bool {
+        state == .data && title.isNotEmpty
+    }
+
+    var calendarName: String? {
+        calendar?.name
+    }
```

Comment:
I think it would be more straightforward to make the `calendarName` `@Published` and assign a value to it when the calendar changes (could be in a `didSet` block) and remove the `@Published` from the `calendar` entity. It looks strange that a variable is published so it will trigger a UI update but the variable is not accessible so there must be something else that actually changes and make something change on the UI.